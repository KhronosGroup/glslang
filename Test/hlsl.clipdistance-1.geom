struct S {
    float4 pos   : SV_Position;
    float  clip  : SV_ClipDistance0;
    float  cull  : SV_CullDistance0;
};

[maxvertexcount(3)]
void main(triangle in float4 pos[3] : SV_Position, 
          triangle in uint VertexID[3] : VertexID,
          inout LineStream<S> OutputStream,
          triangle in float clip[3] : SV_ClipDistance,  // scalar float
          triangle in float cull[3] : SV_CullDistance)  // scalar float
{
    S s;

    s.pos = pos[0];
    s.clip = clip[0];
    s.cull = cull[0];

    OutputStream.Append(s);
}

