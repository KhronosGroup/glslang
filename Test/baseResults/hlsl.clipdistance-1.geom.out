hlsl.clipdistance-1.geom
ERROR: 0:13: '' : unimplemented: clip/cull not currently implemented for this stage 
ERROR: 0:13: '' : unimplemented: clip/cull not currently implemented for this stage 
ERROR: 0:20: '' : unimplemented: clip/cull not currently implemented for this stage 
ERROR: 0:20: '' : unimplemented: clip/cull not currently implemented for this stage 
ERROR: 4 compilation errors.  No code generated.


Shader version: 500
invocations = -1
max_vertices = 3
input primitive = triangles
output primitive = line_strip
ERROR: node is still EOpNull!
0:13  Function Definition: @main(vf4[3];u1[3];struct-S-vf4-f1-f11;f1[3];f1[3]; ( temp void)
0:13    Function Parameters: 
0:13      'pos' ( in 3-element array of 4-component vector of float)
0:13      'VertexID' ( in 3-element array of uint)
0:13      'OutputStream' ( out structure{ temp 4-component vector of float pos,  temp float clip,  temp float cull})
0:13      'clip' ( in 3-element array of float)
0:13      'cull' ( in 3-element array of float)
0:?     Sequence
0:16      move second child to first child ( temp 4-component vector of float)
0:16        pos: direct index for structure ( temp 4-component vector of float)
0:16          's' ( temp structure{ temp 4-component vector of float pos,  temp float clip,  temp float cull})
0:16          Constant:
0:16            0 (const int)
0:16        direct index ( temp 4-component vector of float)
0:16          'pos' ( in 3-element array of 4-component vector of float)
0:16          Constant:
0:16            0 (const int)
0:17      move second child to first child ( temp float)
0:17        clip: direct index for structure ( temp float)
0:17          's' ( temp structure{ temp 4-component vector of float pos,  temp float clip,  temp float cull})
0:17          Constant:
0:17            1 (const int)
0:17        direct index ( temp float)
0:17          'clip' ( in 3-element array of float)
0:17          Constant:
0:17            0 (const int)
0:18      move second child to first child ( temp float)
0:18        cull: direct index for structure ( temp float)
0:18          's' ( temp structure{ temp 4-component vector of float pos,  temp float clip,  temp float cull})
0:18          Constant:
0:18            2 (const int)
0:18        direct index ( temp float)
0:18          'cull' ( in 3-element array of float)
0:18          Constant:
0:18            0 (const int)
0:20      Sequence
0:20        Sequence
0:20          move second child to first child ( temp 4-component vector of float)
0:?             'OutputStream.pos' ( out 4-component vector of float Position)
0:20            pos: direct index for structure ( temp 4-component vector of float)
0:20              's' ( temp structure{ temp 4-component vector of float pos,  temp float clip,  temp float cull})
0:20              Constant:
0:20                0 (const int)
0:20        EmitVertex ( temp void)
0:13  Function Definition: main( ( temp void)
0:13    Function Parameters: 
0:?     Sequence
0:13      move second child to first child ( temp 3-element array of 4-component vector of float)
0:?         'pos' ( temp 3-element array of 4-component vector of float)
0:?         'pos' ( in 3-element array of 4-component vector of float Position)
0:13      move second child to first child ( temp 3-element array of uint)
0:?         'VertexID' ( temp 3-element array of uint)
0:?         'VertexID' (layout( location=0) in 3-element array of uint)
0:13      Function Call: @main(vf4[3];u1[3];struct-S-vf4-f1-f11;f1[3];f1[3]; ( temp void)
0:?         'pos' ( temp 3-element array of 4-component vector of float)
0:?         'VertexID' ( temp 3-element array of uint)
0:?         'OutputStream' ( temp structure{ temp 4-component vector of float pos,  temp float clip,  temp float cull})
0:?         'clip' ( temp 3-element array of float)
0:?         'cull' ( temp 3-element array of float)
0:?   Linker Objects
0:?     'pos' ( in 3-element array of 4-component vector of float Position)
0:?     'VertexID' (layout( location=0) in 3-element array of uint)
0:?     'OutputStream.pos' ( out 4-component vector of float Position)


Linked geometry stage:


Shader version: 500
invocations = 1
max_vertices = 3
input primitive = triangles
output primitive = line_strip
ERROR: node is still EOpNull!
0:13  Function Definition: @main(vf4[3];u1[3];struct-S-vf4-f1-f11;f1[3];f1[3]; ( temp void)
0:13    Function Parameters: 
0:13      'pos' ( in 3-element array of 4-component vector of float)
0:13      'VertexID' ( in 3-element array of uint)
0:13      'OutputStream' ( out structure{ temp 4-component vector of float pos,  temp float clip,  temp float cull})
0:13      'clip' ( in 3-element array of float)
0:13      'cull' ( in 3-element array of float)
0:?     Sequence
0:16      move second child to first child ( temp 4-component vector of float)
0:16        pos: direct index for structure ( temp 4-component vector of float)
0:16          's' ( temp structure{ temp 4-component vector of float pos,  temp float clip,  temp float cull})
0:16          Constant:
0:16            0 (const int)
0:16        direct index ( temp 4-component vector of float)
0:16          'pos' ( in 3-element array of 4-component vector of float)
0:16          Constant:
0:16            0 (const int)
0:17      move second child to first child ( temp float)
0:17        clip: direct index for structure ( temp float)
0:17          's' ( temp structure{ temp 4-component vector of float pos,  temp float clip,  temp float cull})
0:17          Constant:
0:17            1 (const int)
0:17        direct index ( temp float)
0:17          'clip' ( in 3-element array of float)
0:17          Constant:
0:17            0 (const int)
0:18      move second child to first child ( temp float)
0:18        cull: direct index for structure ( temp float)
0:18          's' ( temp structure{ temp 4-component vector of float pos,  temp float clip,  temp float cull})
0:18          Constant:
0:18            2 (const int)
0:18        direct index ( temp float)
0:18          'cull' ( in 3-element array of float)
0:18          Constant:
0:18            0 (const int)
0:20      Sequence
0:20        Sequence
0:20          move second child to first child ( temp 4-component vector of float)
0:?             'OutputStream.pos' ( out 4-component vector of float Position)
0:20            pos: direct index for structure ( temp 4-component vector of float)
0:20              's' ( temp structure{ temp 4-component vector of float pos,  temp float clip,  temp float cull})
0:20              Constant:
0:20                0 (const int)
0:20        EmitVertex ( temp void)
0:13  Function Definition: main( ( temp void)
0:13    Function Parameters: 
0:?     Sequence
0:13      move second child to first child ( temp 3-element array of 4-component vector of float)
0:?         'pos' ( temp 3-element array of 4-component vector of float)
0:?         'pos' ( in 3-element array of 4-component vector of float Position)
0:13      move second child to first child ( temp 3-element array of uint)
0:?         'VertexID' ( temp 3-element array of uint)
0:?         'VertexID' (layout( location=0) in 3-element array of uint)
0:13      Function Call: @main(vf4[3];u1[3];struct-S-vf4-f1-f11;f1[3];f1[3]; ( temp void)
0:?         'pos' ( temp 3-element array of 4-component vector of float)
0:?         'VertexID' ( temp 3-element array of uint)
0:?         'OutputStream' ( temp structure{ temp 4-component vector of float pos,  temp float clip,  temp float cull})
0:?         'clip' ( temp 3-element array of float)
0:?         'cull' ( temp 3-element array of float)
0:?   Linker Objects
0:?     'pos' ( in 3-element array of 4-component vector of float Position)
0:?     'VertexID' (layout( location=0) in 3-element array of uint)
0:?     'OutputStream.pos' ( out 4-component vector of float Position)

SPIR-V is not generated for failed compile or link
